planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2350,6 +2376,30 @@ bool AvoidanceModule::isValidShiftLine(
     }
   }
 
+  // check if the vehicle is in road.
+  {
+    const auto minimum_distance = 0.5 * planner_data_->parameters.vehicle_width +
+                                  parameters_->hard_road_shoulder_margin -
+                                  parameters_->max_departure_from_lane;
+
+    const size_t start_idx = shift_lines.front().start_idx;
+    const size_t end_idx = shift_lines.back().end_idx;
+
+    for (size_t i = start_idx; i <= end_idx; ++i) {
+      const auto p = getPoint(shifter_for_validate.getReferencePath().points.at(i));
+      lanelet::BasicPoint2d basic_point{p.x, p.y};
+
+      const auto shift_length = proposed_shift_path.shift_length.at(i);
+      const auto bound = shift_length > 0.0 ? avoid_data_.left_bound : avoid_data_.right_bound;
+      const auto THRESHOLD = minimum_distance + std::abs(shift_length);
+
+      if (boost::geometry::distance(basic_point, lanelet::utils::to2D(bound)) < THRESHOLD) {
+        RCLCPP_DEBUG_THROTTLE(getLogger(), *clock_, 1000, "road shoulder distance is not enough.");
```

Comment:
It seems this debug info is outputted even if road shoulder distance is big enough?
In my understanding, this condition is met when road shoulder distance is not enough or `max_departure_from_lane` is big?

Reply:
update comment!
```
following avoidance path that is generated from latest new shift line may cause deviation from drivable area.
```